src/libsync/creds/oauth.cpp

```diff
@@ -260,12 +264,33 @@ QByteArray OAuth::generateRandomString(size_t size) const
     return QByteArray(reinterpret_cast<char *>(buffer.data()), static_cast<int>(size * sizeof(quint32))).toBase64(QByteArray::Base64UrlEncoding);
 }
 
+QVariant OAuth::getRequiredField(const QJsonObject &json, const QString &s, QString *error)
+{
+    const auto out = json.constFind(s);
+    if (out == json.constEnd()) {
+        error->append(tr("\tError: Missing field %1\n").arg(s));
+        return QJsonValue();
+    }
+    return *out;
+}
+
+QString OAuth::errorFromJson(const QJsonObject &json)
+{
+    if (json.isEmpty()) {
+        return {};
+    }
+    QString errorFromJson = json[QStringLiteral("error_description")].toString();
+    if (errorFromJson.isEmpty())
```

Comment:
TODO: don't use error_description to identify the error, this is a user targeting string.